pkg/k8s/k8s_service.go

```diff
@@ -149,33 +149,36 @@ func (c *Client) WaitForService(ctx context.Context, name string) error {
 	ticker := time.NewTicker(waitRetry)
 	defer ticker.Stop()
 
-	for {
-		select {
-		case <-ctx.Done():
-			return ErrTimeoutWaitingForServiceReady
-
-		case <-ticker.C:
-			ready, err := c.isServiceReady(ctx, name)
-			if err != nil {
-				return ErrCheckingServiceReady.WithParams(name).Wrap(err)
-			}
-			if !ready {
-				continue
+	for firstIteration := true; ; firstIteration = false {
+		if !firstIteration {
+			select {
+			case <-ctx.Done():
+				return ErrTimeoutWaitingForServiceReady
+			case <-ticker.C:
+				// Wait for the next tick before proceeding
 			}
+		}
```

Comment:
if it isn't important to use a ticker, i.e. checking on a strict interval, then we could just use `time.After` and avoid using this `firstIteration` variable. 
The additional benefit of the `time.After` is that we could have custom intervals based on the retry reason. 

```suggestion
        retryInterval := 0
	for  {
			select {
			case <-ctx.Done():
				return ErrTimeoutWaitingForServiceReady
			case <-time.After(retryInterval):
			        // Reset to default interval
			        retryInterval = waitRetry
			}
```

Reply:
we can also use `time.After` but still without `firstIteration` in this particular case, it waits for the `retryInterval` and then checks is the service is ready. The point of having that `firstIteration` is to check it first and then start the waiting.

Reply:
this is why my example starts with `retryInterval = 0`

Reply:
got it, nice trick